src/pydocstyle/checker.py

```diff
@@ -390,6 +411,174 @@ def check_starts_with_this(self, function, docstring):
             if first_word.lower() == 'this':
                 return violations.D404()
 
+    @staticmethod
+    def _get_leading_words(line):
+        """Return any leading set of words from `line`.
+
+        For example, if `line` is "  Hello world!!!", returns "Hello world".
+        """
+        result = re("[A-Za-z ]+").match(line.strip())
+        if result is not None:
+            return result.group()
+
+    @staticmethod
+    def _is_a_docstring_section(context):
+        """Check if the suspected line is really a section.
+
+        This is done by checking the following conditions:
+        * Does the current line has a suffix after the suspected section name?
+        * Is the previous line not empty?
+        * Does the previous line end with a punctuation mark?
+
+        If so, this is probably not a real section name. For example:
+            '''Title.
+
+            Some part of the docstring that specifies what the function
+            returns. <----- Not a real section name.
+            '''
+        """
+        section_name_suffix = context.line.lstrip(context.section_name).strip()
+
+        punctuation = [',', ';', '.', '-', '\\', '/', ']', '}', ')']
+        prev_line_ends_with_punctuation = \
+            any(context.previous_line.strip().endswith(x) for x in punctuation)
+
+        return not (section_name_suffix != '' and not
+                    prev_line_ends_with_punctuation and not
+                    context.previous_line.strip() == '')
+
+    @classmethod
+    def _check_section_underline(cls, section_name, context, indentation):
+        """D4{07,08,09,10}, D215: Section underline checks.
+
+        Check for correct formatting for docstring sections. Checks that:
+            * The line that follows the section name contains
+              dashes (D40{7,8}).
+            * The amount of dashes is equal to the length of the section
+              name (D409).
+            * The line that follows the section header (with or without dashes)
+              is empty (D410).
+            * The indentation of the dashed line is equal to the docstring's
+              indentation (D215).
+        """
+        dash_line_found = False
+        next_non_empty_line_offset = 0
+
+        for line in context.following_lines:
+            line_set = ''.join(set(line.strip()))
```

Comment:
I used it anywhere i tested that. This line is not the case, i want to see that it has only dashes in it.